70_FLIPFluids/gpuRadixSort.h

```diff
@@ -0,0 +1,470 @@
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			},
+            {
+				.binding=2,
+				.type=nbl::asset::IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
+				.createFlags=IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			},
+			{
+				.binding=3,
+				.type=nbl::asset::IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
+				.createFlags=IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			},
+			{
+				.binding=4,
+				.type=nbl::asset::IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
+				.createFlags=IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			}
+		};
+		smart_refctd_ptr<IGPUDescriptorSetLayout> sortDsLayout = device->createDescriptorSetLayout(sortBindingsSet);
+
+		smart_refctd_ptr<nbl::video::IGPUPipelineLayout> sortPplnLayout = device->createPipelineLayout({}, nullptr, smart_refctd_ptr(sortDsLayout), nullptr, nullptr);
+
+		// create pipelines
+		auto compileShader = [&](const std::string& path) -> smart_refctd_ptr<IGPUShader>
+			{
+				IAssetLoader::SAssetLoadParams lp = {};
+				lp.logger = logger.get();
+				lp.workingDirectory = "";
+				auto bundle = assetManager->getAsset(path, lp);
+				const auto assets = bundle.getContents();
+				assert(assets.size() == 1);
+				smart_refctd_ptr<ICPUShader> shaderSrc = IAsset::castDown<ICPUShader>(assets[0]);
+				shaderSrc->setShaderStage(IShader::E_SHADER_STAGE::ESS_COMPUTE);
+				if (!shaderSrc)
+					return nullptr;
+
+				return m_device->createShader(shaderSrc.get());
+			};
+
+		{
+            smart_refctd_ptr<video::IGPUShader> shader = compileShader("app_resources/compute/radix_sort/buildHistogram.comp.hlsl");
+
+            IGPUComputePipeline::SCreationParams params = {};
+			params.layout = histogramPplnLayout.get();
+            params.shader.shader = shader.get();
+
+            m_device->createComputePipelines(nullptr, {&params, 1}, &m_buildHistogramPipeline);
+        }
+		{
+            smart_refctd_ptr<video::IGPUShader> shader = compileShader("app_resources/compute/radix_sort/scanSpine.comp.hlsl");
+
+            IGPUComputePipeline::SCreationParams params = {};
+			params.layout = spinePplnLayout.get();
+            params.shader.shader = shader.get();
+
+            m_device->createComputePipelines(nullptr, {&params, 1}, &m_scanSpinePipeline);
+        }
+        {
+            smart_refctd_ptr<video::IGPUShader> shader = compileShader("app_resources/compute/radix_sort/radixSort.comp.hlsl");
+
+            IGPUComputePipeline::SCreationParams params = {};
+			params.layout = sortPplnLayout.get();
+            params.shader.shader = shader.get();
+
+            m_device->createComputePipelines(nullptr, {&params, 1}, &m_radixSortPipeline);
```

Comment:
you're missing `computeFullSubgroups` and subgroupSize.... this is very importantm, without these `workgroup::reduce` are UB